extras/create_rcsinfo.sh

```diff
@@ -6,7 +6,7 @@
 
 # default value, if git is not installed or if the source directory is
 # not under version control
-version="v2.10.0"
+version="v3.1.0-SNAPSHOT"
```

Comment:
Hmm, making a file-creator script creator. Sounds complex. I guess it could grep out of config.h as well, or use a c preprocessor to extract it. A bit complex as well. Maybe source a very small file that does the version definition made by cmake?

Or maybe just stick this all in cmake and just add an operator that pulls it from config.h?

Tracking version control information inside of projects is always painful, and I tend to think not so useful without the build infrastructure, and if you have the build infrastructure, you already have that information in the tree.

Maybe we only want to track "Is this release version X" or "You built this yourself, you should know how to give us the hash"?